webapp/src/components/modals/attach_issue/attach_issue.jsx

```diff
@@ -0,0 +1,241 @@
+        const description = e.target.value;
+        const {fields} = this.state;
+        const nFields = {
+            ...fields,
+            description,
+        };
+
+        this.setState({fields: nFields});
+    };
+
+    handleProjectChange = (id, value) => {
+        const fields = {...this.state.fields};
+        const projectKey = value;
+        fields.project = {
+            key: value,
+        };
+
+        this.setState({
+            projectKey,
+            fields,
+        });
+    }
+
+    handleIssueKeyChange = (id, value) => {
+        const fields = {...this.state.fields};
+        const issueKey = value;
+
+        fields.ID = value;
+        this.setState({
+            issueKey,
+            fields,
+        });
+    }
+
+    handleFieldChange = (id, value) => {
+        const fields = {...this.state.fields};
+        fields[id] = value;
+        this.setState({
+            fields,
+        });
+    }
+
+    render() {
+        const {post, visible, theme, jiraIssueMetadata} = this.props;
+        const {error, submitting} = this.state;
+        const style = getStyle(theme);
+
+        if (!visible) {
+            return null;
+        }
+
+        let component;
+        if (error) {
+            console.error('render error', error); //eslint-disable-line no-console
+        }
+
+        if (!post || !jiraIssueMetadata || !jiraIssueMetadata.projects) {
+            component = <Loading/>;
+        } else {
+            const projectOptions = getProjectValues(jiraIssueMetadata);
+            component = (
+                <div style={style.modal}>
+                    <ReactSelectSetting
+                        name={'project'}
+                        label={'Project'}
+                        required={true}
+                        onChange={this.handleProjectChange}
+                        placeholder={'Select the project containing the issue you wish to attach the message to.'}
+                        options={projectOptions}
+                        isMuli={false}
+                        key={'LT'}
+                        value={projectOptions.filter((option) => option.value === this.state.projectKey)}
+                    />
+                    <Input
+                        key='key'
+                        id='issueKey'
+                        placeholder={'Enter the issue key, e.g. EXT-20'}
+                        label='Issue Key'
+                        type='input'
+                        onChange={this.handleIssueKeyChange}
+                        required={true}
+                        disabled={false}
+                    />
+                    <Input
+                        label='Message Attached to Jira Issue'
+                        type='textarea'
+                        isDisabled={true}
+                        value={this.state.fields.description}
+                        disabled={false}
+                    />
+                    <JiraFields
```

Comment:
What are the fields for?

Reply:
These aren't getting used and I can remove them.  In `create issue`, this was used to populate issue types, from using the metadata, in a more programmatic way.  I just used two <input /> components to 1) allow user to input an issue Key (ie. EXT-20) and 2) show the comment message from MM.  (with disabled prop so cannot be edited)